File: Data Structures & Algorithms/sliding-window-maximum/submission-10.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        deque<int> dq;
        int n = nums.size();
        vector<int> res;
        for (int j = 0; j < n; ++j) {
            while (!dq.empty() && nums[dq.back()] <= nums[j]) {
                dq.pop_back();
            }
            dq.push_back(j);
            if (j >= k - 1) {
                res.push_back(nums[dq.front()]);
                int i = j - k + 1;
                if (dq[0] <= i) {
                    dq.pop_front();
                }
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-10.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0 

        s = set(nums)

        for num in nums:
            if num - 1 in s:
                continue
            val = num
            curr = 0
            while val in s:
                curr += 1
                val += 1
            res = max(res, curr)
        
        return res
